lms/djangoapps/branding/api.py

```diff
@@ -0,0 +1,289 @@
+        "social_links": [
+            {
+                "name": "facebook",
+                "title": "Facebook",
+                "url": "http://www.facebook.com/example",
+                "icon-class": "fa-facebook-square"
+            },
+            ...
+        ],
+        "navigation_links": [
+            {
+                "name": "about",
+                "title": "About",
+                "url": "http://www.example.com/about.html"
+            },
+            ...
+        ],
+        "mobile_links": [
+            {
+                "name": "apple",
+                "title": "Apple",
+                "url": "http://store.apple.com/example_app"
+                "image": "http://example.com/static/apple_logo.png"
+            },
+            ...
+        ],
+        "legal_links": [
+            {
+                "url": "http://example.com/terms-of-service.html",
+                "name": "terms_of_service",
+                "title': "Terms of Service"
+            },
+            # ...
+        ],
+        "openedx_link": {
+            "url": "http://open.edx.org",
+            "title": "Powered by Open edX",
+            "image": "http://example.com/openedx.png"
+        }
+    }
+
+    """
+    return {
+        "copyright": _footer_copyright(),
+        "logo_image": _footer_logo_img(is_secure),
+        "social_links": _footer_social_links(),
+        "navigation_links": _footer_navigation_links(),
+        "mobile_links": _footer_mobile_links(is_secure),
+        "legal_links": _footer_legal_links(),
+        "openedx_link": _footer_openedx_link(),
+    }
+
+
+def _footer_copyright():
+    """Return the copyright to display in the footer.
+
+    Returns: unicode
+
+    """
+    org_name = (
+        "edX Inc" if settings.FEATURES.get('IS_EDX_DOMAIN', False)
+        else microsite.get_value('PLATFORM_NAME', settings.PLATFORM_NAME)
+    )
+
+    # Translators: 'EdX', 'edX', and 'Open edX' are trademarks of 'edX Inc.'.
+    # Please do not translate any of these trademarks and company names.
+    return _(
+        u"\u00A9 {org_name}.  All rights reserved except where noted.  "
+        u"EdX, Open edX and the edX and OpenEdX logos are registered trademarks "
+        u"or trademarks of edX Inc."
+    ).format(org_name=org_name)
+
+
+def _footer_openedx_link():
+    """Return the image link for "powered by OpenEdX".
+
+    Args:
+        is_secure (bool): Whether the request is using TLS.
+
+    Returns: dict
+
+    """
+    # Translators: 'Open edX' is a brand, please keep this untranslated.
+    # See http://openedx.org for more information.
+    title = _("Powered by Open edX")
+    return {
+        "url": settings.FOOTER_OPENEDX_URL,
+        "title": title,
+        "image": settings.FOOTER_OPENEDX_LOGO_IMAGE,
+    }
+
+
+def _footer_social_links():
+    """Return the social media links to display in the footer.
+
+    Returns: list
+
+    """
+    links = []
+    for social_name in settings.SOCIAL_MEDIA_FOOTER_NAMES:
```

Comment:
this may be beyond the scope of this PR but is there a reason why the names, urls, and display attributes are stored in separate configuration settings?  It seems somewhat fragile, while using a single dictionary might alleviate the need to fall back on empty defaults for hrefs, css class names, etc